### DISCUSSION TITLE
Loading LaTeX style files

### DISCUSSION THREAD

**Discussion by BruceWestbury** [executablebooks/meta#164] [Q&A]
When I use LaTeX I usually load packages using \usepackage{...} in the preamble.

I believe I can do something similar in MyST by putting something in _config.yml.
Can anyone tell me what I need to include?

**Comment by choldgraf** [comment]
cc @mmcky who might have ideas on LaTeX via Sphinx!

**Comment by AakashGfude** [comment]
Hey, @BruceWestbury at present your best bet might be to use the `preamble` key of the `latex_elements` config variable,  in `_config.yml`. Something like this: https://github.com/executablebooks/jupyter-book/blob/master/docs/_config.yml#L60 . Hope this might help.
Also, since, you are from a latex background. we would love to hear more about what other features related to latex you constantly use and want in `jupyter-book`. :)

**Comment by BruceWestbury** [comment]
Thanks. I found that and it worked for me. However this is for
\newcommand{...} whereas I was asking for a way to load a style file
e.g. \usepackage{amssymb} will load amssymb.sty (the AMS symbols package).

I think this is possible. This page
https://www.sphinx-doc.org/en/master/latex.html has a line

\usepackage[titles]{tocloft}

This page https://runawayhorse001.github.io/SphinxGithub/sphinx.html
also has \usepackage in 4. Image math formula preamble

BTW I found it strange that I had to put in each macro twice. Also I think
\N for \mathbb{N} is now standard but probably
in a style file.

I would be happy to discuss my experiences with you. This is my first
jupyter-book so I am on a steep learning curve.

**Reply by mmcky** [reply]
Hi @BruceWestbury -- once you add the `\usepackage` to your preamble -- you should be able to also specify the `sty` file  as [latex_additional_files](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-latex_additional_files) in the `sphinx` config section of the `_config.yml` file. 

```yml
sphinx
  config:
    latex_additional_files = ["mystyle.sty"]
```

I haven't actually used this `config` variable yet -- so please let me know how you go.

**Comment by BruceWestbury** [comment]
Could you please give me the lines to include? both to add \usepackage to my preamble and then to add my style files.

As I understand it, the sphinx documentation and your lines are in ReST whereas jupyter-book uses MyST.
Please clarify if I am mistaken. In the above should it be `sphinx:`?

This is what I currently have

`sphinx:
  config:
    mathjax_config:
      TeX:
        Macros:
          "bN": "\\mathbb{N}"
          "bQ": "\\mathbb{Q}"
          "fS": "\\mathfrak{S}"
          "ST": "\\mathrm{ST}"
          "GT": "\\mathrm{GT}"
    latex_elements:
        preamble:
          \newcommand\bN{\mathbb{N}}
          \newcommand\bQ{\mathbb{Q}}
          \newcommand\fS{\mathfrak{S}}
          \newcommand\ST{\mathrm{ST}}
          \newcommand\GT{\mathrm{GT}}`

[Sorry, this should have line breaks.]

I am guessing that the reason I have to put each macro in twice is that one is for mathjax which is used to build the html and the other is for the latex engine used to build pdf. If this is correct, then will I get style files to work with mathjax or just the latex engine?

**Reply by BruceWestbury** [reply]
I have just looked into this and it seems that MathJax gives a set of LaTeX commands but has no facility for adding to the list.
As I want to be able to build both html and pdf, it seems I will have to work with the LaTeX commands that are implemented in MathJax (and macros).

**Reply by mmcky** [reply]
sorry for delay on this @BruceWestbury -- not sure what you quite mean here `no facility for adding to the list`. Can you clarify and I can try and help out. 

Did you get the syntax above to work in your `_config.yml`?
